libraries/AP_Generator/AP_Generator_RichenPower.cpp

```diff
@@ -326,25 +395,38 @@ bool AP_Generator_RichenPower::healthy() const
 //send generator status
 void AP_Generator_RichenPower::send_generator_status(const GCS_MAVLINK &channel)
 {
-    uint64_t status = 0;
-    switch (last_reading.mode) {
-    case Mode::IDLE:
-        break;
-    case Mode::RUN:
-        status |= MAV_GENERATOR_STATUS_FLAG_GENERATING;
-        break;
-    case Mode::CHARGE:
-        status |= MAV_GENERATOR_STATUS_FLAG_GENERATING;
-        status |= MAV_GENERATOR_STATUS_FLAG_CHARGING;
-        break;
-    case Mode::BALANCE:
-        status |= MAV_GENERATOR_STATUS_FLAG_GENERATING;
-        status |= MAV_GENERATOR_STATUS_FLAG_CHARGING;
-        break;
+    if (last_reading_ms == 0) {
+        // nothing to report
+        return;
     }
 
+    uint64_t status = 0;
```

Comment:
absolutely not your fault I think but I think 64bits is a bit much for the status.  Surely 32 would have been more than enough for a single device.